src/server/routes/editor.tsx

```diff
@@ -531,19 +531,32 @@ router.get('/:id/achievements', async (req, res, next) => {
 		.where('editor_id', userId)
 		.fetchAll({require: false});
 
+	const achievementCol = new AchievementUnlock()
+		.where('editor_id', userId)
+		.where('profile_rank', '<=', '3')
+		.query((qb) => qb.limit(3))
+		.orderBy('profile_rank', 'ASC')
+		.fetchAll({
+			require: false,
+			withRelated: ['achievement']
+		});
+
 	const achievementTypesPromise = new AchievementType()
 		.orderBy('id', 'ASC')
 		.fetchAll();
 
 	const [unlocks, editorJSON, achievementTypes] = await Promise.all([
-		unlocksPromise, editorJSONPromise, achievementTypesPromise
+		unlocksPromise, editorJSONPromise, achievementTypesPromise, achievementCol
 	]);
 
 	const achievementJSON =
 		await setAchievementUnlockedField(achievementTypes, unlocks, userId, res.app.locals.orm);
 
+	const currAchievementJSON = achievementColToEditorGetJSON(achievementCol);
```

Comment:
`achievementCol` here is the promise itself rather than the values we get after awaiting it.
I would recommend renaming it to `achievementColPromise` like we do for the other ones, and you'll need to add a new `achievementCol` on line 548 where we destructure the results of Promise.all: 
`const [unlocks, editorJSON, achievementTypes, achievementCol] = await Promise.all(`

I suspect currently the code is not working.